components/script/dom/bindings/refcounted.rs

```diff
@@ -161,7 +160,7 @@ impl TrustedPromise {
 pub struct Trusted<T: DomObject> {
     /// A pointer to the Rust DOM object of type T, but void to allow
     /// sending `Trusted<T>` between threads, regardless of T's sendability.
-    refcount: Arc<TrustedReference>,
+    refcount: Rc<TrustedReference>,
```

Comment:
Given the comment and usage I am pretty sure we need to keep `Arc` here, right @jdm?

Reply:
> Given the comment and usage I am pretty sure we need to keep `Arc` here, right @jdm?

hello! there was a discussion here on the changes to be done: https://github.com/servo/servo/issues/33726 please go through them and let me know if this is the right way about it.

Reply:
The problem is that we impl `Send` for `Trusted` (and `Clone` too): https://github.com/servo/servo/blob/ff6523c37e132605c0945bdc49f15b75b0566d2e/components/script/dom/bindings/refcounted.rs#L169, so if we send Trusted to another thread it will not lower ref count on drop in other thread, so we will keep it alive permanently in main thread.

Reply:
You're right. Switching to Rc here looks like we risk racing modifying the refcount value between threads, so this change is not safe.

Reply:
hello @jdm @sagudev , could you please let me know if this warning is best ignored or should i be implementing either a Send and Sync for TrustedReference or Wrap TrustedReference in a Mutex ? alsoo i need help with deciding the implementation of a similar PR https://github.com/servo/servo/pull/33718 I am still trying to understand the larger context of their usage, any insights on that would be super helpful!!

Reply:
I think we should just ignore (allow) clippy lint (probably for whole script crate).

Reply:
We are very close to fixing these issues in the script crate. I don't think we should ignore errors, because it will cause many more to appear.

Reply:
:sob:

Reply:
> When I said ignore I meant allow as some clippy errors are simply wrong (sometimes). This is known, so that's why there are not in rustc.

Ah, that makes sense. Thanks for the clarification.

Reply:
Agreed with sagudev. I'm going to close this PR, since this is no longer the desired fix for this particular warning instance. Sorry for sending you down the wrong path!